### PR TITLE
Feature/10224 theme picker tracking

### LINKED ISSUE #10382: Light weight storefront feature tracking
Screenshot 2023-12-13 at 16 26 44

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #10382

### Description

Apply the following tracking plan for theme picker feature:

Screenshot 2023-12-13 at 16 26 44

### Testing instructions

See the above table for reference. Follow the steps below and check for the tracked events in the logcat

**From settings**
1. Open Theme picker screen from settings: `Tracked: theme_picker_screen_displayed, Properties: {"source":"settings"`
2. Select any theme from the picker to open the preview: `Tracked: theme_picker_theme_selected, Properties: {"theme":"Attar"`
3. Preview is opened: `Tracked: theme_preview_screen_displayed`
4. Change theme page: `theme_preview_page_selected, Properties: {"page":"My Account"`
5. Change device form factor layout: `Tracked: theme_preview_layout_selected, Properties: {"layout":"mobile"`
6. Disable the internet and tap on "Use this theme": `Tracked: theme_preview_start_with_theme_button_tapped, Properties: {"theme":"Tsubaki"` and  `theme_installation_failed, Properties: {"theme":"Tsubaki"`
7. Enable internet connection again and tap on "Use this theme": `Tracked: theme_installation_completed, Properties: {"theme":"Tsubaki"`

**From store creation**
1. Open theme picker from store creation and check: `theme_picker_screen_displayed, Properties: {"source":"profiler"` 
**Note:** About the `source` value for this case, ' profiler`, do you think it might make more sense to name it `store_creation`?  Imo the theme picker is not a profiler question/step. It is just another screen in the store creation flow. Wdyt @hafizrahman ?

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/themes/ThemePickerViewModel.kt [hunk 1]
```diff
@@ -71,7 +72,7 @@ class ThemePickerViewModel @Inject constructor(
                         .filter { theme -> theme.demoUrl != null }
                         .map { theme ->
                             CarouselItem.Theme(
-                                themeId = theme.id,
+                                uri = theme.id,
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/themes/ThemePickerViewModel.kt]: @JorgeMucientes can you double check this? this holds the theme `id`, why it was renamed to `uri`.

Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/themes/ThemePickerViewModel.kt]: Looking at the other changes, I think it was just an accidental one, so reverting it should be straightforward.

Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/themes/ThemePickerViewModel.kt]: Good catch! Yeah, this was definitely a mistake. Reverted

**Comment by hichamboushaba** [comment]
@JorgeMucientes I reviewed the existing tracks in the PR, I will review the remaining once you add them since #10294 is merged now.

**Comment by JorgeMucientes** [comment]
Thank you @hichamboushaba. I've just merged `trunk` and added the missing tracking. Ready for a final check

**Review by hichamboushaba — APPROVED** [review]
Works well, nice work @JorgeMucientes

**Comment by hafizrahman** [comment]
> Imo the theme picker is not a profiler question/step. It is just another screen in the store creation flow. Wdyt @hafizrahman ?

Sorry I missed this ping. You are correct that `store_creation` would've been semantically more correct. Seeing that this is already merged, if you want to change it on beta / next release please let me know and I'll update the PT.

**Pull request merged**